metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/DepublishedRecordController.java

```diff
@@ -0,0 +1,163 @@
+    final MetisUser metisUser = authenticationClient.getUserByAccessTokenInHeader(authorization);
+    final int added = depublishedRecordService
+            .addRecordsToBeDepublished(metisUser, datasetId, recordIdsInSeparateLines);
+    LOGGER.info("{} Depublished records added to dataset with datasetId: {}", added, datasetId);
+  }
+
+  /**
+   * Adds a list of depublished records to the dataset - the version for a multipart file.
+   *
+   * @param authorization the HTTP Authorization header, in the form of a Bearer Access Token.
+   * @param datasetId The dataset ID to which the depublished records belong.
+   * @param recordIdsFile The file containing the record IDs in separate lines.
+   * @throws GenericMetisException which can be one of:
+   * <ul>
+   * <li>{@link NoDatasetFoundException} if the dataset for datasetId was not found.</li>
+   * <li>{@link UserUnauthorizedException} if the user is unauthorized</li>
+   * <li>{@link BadContentException} if some content or the operation were invalid</li>
+   * </ul>
+   */
+  @PostMapping(value = RestEndpoints.DEPUBLISHED_RECORDS_DATASETID, consumes = {
+          MediaType.MULTIPART_FORM_DATA_VALUE})
+  @ResponseStatus(HttpStatus.CREATED)
+  public void createDepublishedRecords(
+          @RequestHeader("Authorization") String authorization,
+          @PathVariable("datasetId") String datasetId,
+          @RequestPart("depublicationFile") MultipartFile recordIdsFile
+  ) throws GenericMetisException, IOException {
+    createDepublishedRecords(authorization, datasetId,
+            new String(recordIdsFile.getBytes(), StandardCharsets.UTF_8));
+  }
+
+  /**
+   * Retrieve the list of depublished records for a specific dataset.
+   *
+   * @param authorization the HTTP Authorization header, in the form of a Bearer Access Token.
+   * @param datasetId The ID of the dataset for which to retrieve the records.
+   * @param page The page to retrieve.
+   * @param sortFieldString The field on which to sort.
+   * @param sortDirectionString The direction in which to sort.
+   * @param searchQuery Search query for the record ID.
+   * @return A list of records.
+   * @throws GenericMetisException which can be one of:
+   * <ul>
+   * <li>{@link NoDatasetFoundException} if the dataset for datasetId was not found.</li>
+   * <li>{@link UserUnauthorizedException} if the user is unauthorized</li>
+   * </ul>
+   */
+  @GetMapping(value = RestEndpoints.DEPUBLISHED_RECORDS_DATASETID, produces = {
+          MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public ResponseListWrapper<DepublishedRecordView> getDepublishedRecords(
+          @RequestHeader("Authorization") String authorization,
+          @PathVariable("datasetId") String datasetId,
+          @RequestParam(value = "page", defaultValue = "0") int page,
+          @RequestParam(value = "sortField", defaultValue = "") String sortFieldString,
```

Comment:
I know that :) But in this case the sorting parameter from the frontend is not exactly equal to the enum names (for instance it is not even upper case). 
Rather than impose this on the frontend I thought It would be better to keep the exact input values uncoupled from our internal java code. 
How do you feel about this?

Reply:
I feel that since we have control of the frontend we should enforce the same values..
There is no need to have different values, that could perhaps be confusing..
We do similarly for other calls for workflow executions for example I believe

Reply:
OK, I'll change it.